generators/entity-server/templates/src/main/java/package/common/patch_template.ejs

```diff
@@ -25,42 +25,38 @@ const returnPrefix = (isService) ? 'return' : returnType + '<' + instanceType +
 const mapper = entityInstance  + 'Mapper';
 _%>
 <%_ if(viaService) { _%>
-  <%_ if(!reactive) { _%>
+<%_ if(!reactive) { _%>
     Optional<<%= instanceType %>> result = <%= entityInstance %>Service.partialUpdate(<%= instanceName %>);
-  <%_ } else { _%>
+<%_ } else { _%>
     Mono<<%= instanceType %>> result = <%= entityInstance %>Service.partialUpdate(<%= instanceName %>);
-  <%_ } _%>
+<%_ } _%>
 <%_ } else { %>
-<%- returnPrefix %> <%= entityInstance %>Repository.findById(<%= instanceName %>.get<%= primaryKey.nameCapitalized %>())
-   .map(existing<%= entityClass %> -> {
-  <%_ if(dtoMapstruct) { _%>
-       <%= mapper %>.partialUpdate(existing<%= entityClass %>, <%= instanceName %>);
-  <%_ } else { _%>
+    <%- returnPrefix %> <%= entityInstance %>Repository.findById(<%= instanceName %>.get<%= primaryKey.nameCapitalized %>())
+    .map(existing<%= entityClass %> -> {
+    <%_ if(dtoMapstruct) { _%>
+        <%= mapper %>.partialUpdate(existing<%= entityClass %>, <%= instanceName %>);
+    <%_ } else { _%>
     <%_ for (const field of fields.filter(field => !field.id && !field.transient)) { _%>
-    if (<%= instanceName %>.get<%= field.fieldInJavaBeanMethod %>() != null) {
+        if (<%= instanceName %>.get<%= field.fieldInJavaBeanMethod %>() != null) {
         existing<%= entityClass %>.set<%= field.fieldInJavaBeanMethod %>(<%= instanceName %>.get<%= field.fieldInJavaBeanMethod %>());
-    }
-      <%_ if (field.fieldWithContentType) { _%>
-    if (<%= instanceName %>.get<%= field.fieldInJavaBeanMethod %>ContentType() != null) {
-        existing<%= entityClass %>.set<%= field.fieldInJavaBeanMethod %>ContentType(<%= instanceName %>.get<%= field.fieldInJavaBeanMethod %>ContentType());
-    }
-      <%_ } _%>
+        }
+        <%_ if (field.fieldWithContentType) { _%>
+            if (<%= instanceName %>.get<%= field.fieldInJavaBeanMethod %>ContentType() != null) {
+            existing<%= entityClass %>.set<%= field.fieldInJavaBeanMethod %>ContentType(<%= instanceName %>.get<%= field.fieldInJavaBeanMethod %>ContentType());
+            }
+        <%_ } _%>
     <%_ } %>
-  <% } %>
+    <%_ } _%>
     return existing<%= entityClass %>;
     })
     .<%= mapOrFlatMap %>(<%= entityInstance %>Repository::save)
-  <%_ if (searchEngineElasticsearch) { _%>
-    .<%= mapOrFlatMap %>(saved<%= entityClass %> -> {
+    <%_ if (searchEngineElasticsearch) { _%>
+        .<%= mapOrFlatMap %>(<% if(reactive) { %><%= entityInstance %>SearchRepository::save<% } else { %>saved<%= entityClass %> -> {
```

Comment:
Unfortunately it not as simple as that: elastic won't persist the relationships because they are lazy because of hibernate. Are you really sure it will work well with reactive and realtions?

Reply:
Wouldn't we have the same for non reactive apps? It also just calls `save`.

Reply:
With reactive there is no hibernate (and no lazy relationships). But indeed, there is the limitation that related entities are not automatically persisted like with JPA (and non reactive apps). I think, this does not only apply to `partialUpdate` but also applies to `create` and has to do with how reactive is working.

It should also be noted that with reactive the REST-endpoints for `create`, `update` and `patch` (only after committing this PR) return the result of the SearchRepository::save whereas with non reactive the same REST-endpoints return the result of Repository::save. I prefer the latter having the database as the source of truth and not the search-technology. But if this should also be changed it would be a slightly bigger scope than just the fix related to this PR.

Reply:
Isn't it another solution?
like for example something which is a little bit less 'FP' but returning the sql Mono/Flux?
```
<%_ if (searchEngineElasticsearch) { _%>
        .<%= mapOrFlatMap %>(<% if(reactive) { %>sqlResult -> <%= entityInstance %>SearchRepository.save(sqlResult).map(esResult -> sqlResult)<% } else { %>saved<%= entityClass %> -> {
```

Reply:
@Tcharl Yes, I would also prefer your solution. The last `.map(esResult -> sqlResult)` should be changed to `.then(Mono.just(sqlResult))`